In [1]:
import pandas as pd
import csv

In [3]:
vine_df = pd.read_csv('vine_table.csv')
vine_df.head()

review_id  star_rating  helpful_votes  total_votes vine  \
0  R3VR960AHLFKDV            4              0            0    N   
1  R16LGVMFKIUT0G            5              0            0    N   
2  R1AIMEEPYHMOE4            5              1            1    N   
3  R1892CCSZWZ9SR            3              0            0    N   
4  R285P679YWVKD1            3              0            0    N   

  verified_purchase  
0                 Y  
1                 Y  
2                 Y  
3                 Y  
4                 N

Total Votes

In [5]:
total_votes_filtered = vine_df[vine_df['total_votes'] >= 20]
total_votes_filtered.head()

review_id  star_rating  helpful_votes  total_votes vine  \
73    RL8D0KJ0J9L0O            5            152          165    N   
157  R1BEINAIQFBRJC            5             21           23    N   
190  R2L59KIJH302P9            4             26           26    N   
221   RR99CPG695T0I            5            215          248    N   
237  R1XQNKKUPCMWVO            5             43           44    N   

    verified_purchase  
73                  Y  
157                 Y  
190                 Y  
221                 N  
237                 Y

Helpful Votes

In [6]:
helpful_votes_filtered = total_votes_filtered[total_votes_filtered['helpful_votes']/total_votes_filtered['total_votes'] >= 0.5]
helpful_votes_filtered.head()

review_id  star_rating  helpful_votes  total_votes vine  \
73    RL8D0KJ0J9L0O            5            152          165    N   
157  R1BEINAIQFBRJC            5             21           23    N   
190  R2L59KIJH302P9            4             26           26    N   
221   RR99CPG695T0I            5            215          248    N   
237  R1XQNKKUPCMWVO            5             43           44    N   

    verified_purchase  
73                  Y  
157                 Y  
190                 Y  
221                 N  
237                 Y

Filter for reviews that are part of the Vine program (Paid)

In [7]:
vine_program_reviews = helpful_votes_filtered[helpful_votes_filtered['vine'] == 'Y']
vine_program_reviews.head()

review_id  star_rating  helpful_votes  total_votes vine  \
5408   R2BQOD1R0228FN            3             17           26    Y   
8246    RC31RUPFOHBHQ            4            102          117    Y   
12219   REN3N1WITLF1Y            5             33           37    Y   
13330   R71RZQ9UZVG47            4             38           47    Y   
16942  R38NMQBH88HLM6            4             18           24    Y   

      verified_purchase  
5408                  N  
8246                  N  
12219                 N  
13330                 N  
16942                 N

Filter for reviews that are NOT part of the Vine program

In [8]:
not_vine_reviews = helpful_votes_filtered[helpful_votes_filtered['vine'] == 'N']
not_vine_reviews.head()

review_id  star_rating  helpful_votes  total_votes vine  \
73    RL8D0KJ0J9L0O            5            152          165    N   
157  R1BEINAIQFBRJC            5             21           23    N   
190  R2L59KIJH302P9            4             26           26    N   
221   RR99CPG695T0I            5            215          248    N   
237  R1XQNKKUPCMWVO            5             43           44    N   

    verified_purchase  
73                  Y  
157                 Y  
190                 Y  
221                 N  
237                 Y

5 Star Review Analysis

In [9]:
count_vine = len(vine_program_reviews)
count_vine

136

In [10]:
count_not_vine = len(not_vine_reviews)
count_not_vine

18019

In [12]:
rating_vine = len(vine_program_reviews[vine_program_reviews['star_rating']==5])
rating_vine

74

In [15]:
rating_not_vine = len(not_vine_reviews[not_vine_reviews['star_rating']==5])
rating_not_vine

8482

In [16]:
rating_vine/count_vine

0.5441176470588235

In [17]:
rating_not_vine/count_not_vine

0.47072534546867195

Paid review Summary: 

review count: 136, 
5 star reviews: 74 ,
Percentage of 5 star reviews: 54.4% ,

Unpaid review summary:
review count: 18019,
5 star reviews: 8482,
Percentage of 5 star reviews: 47%,